config/initializers/_service_providers.rb

```diff
@@ -2,3 +2,7 @@
                     fetch(Rails.env, {})
 
 VALID_SERVICE_PROVIDERS = JSON.parse(Figaro.env.valid_service_providers)
+
+# merge from dashboard
+require 'feature_management'
+ServiceProviderUpdater.new.run if FeatureManagement.use_dashboard_service_providers?
```

Comment:
As written, this appears to be a synchronous request in the starting of the app -- I would very much like to avoid this as I think it behaves very poorly in a failure scenario like the dashboard being down or deployed.

I think we should move this to a background thread or something, and then see if we can have the app mark itself unhealthy to our load balancer until it gets acceptable data?

For the time being (or the life of this PR) I don't think we need the unhealhty part, but I feel very strongly about not having a synchronous web request in this part of theapp.